File: test/wordfilter_test.js
'use strict';

var wordfilter = require('../lib/wordfilter.js');

/*
  ======== A Handy Little Nodeunit Reference ========
  https://github.com/caolan/nodeunit

  Test methods:
    test.expect(numAssertions)
    test.done()
  Test assertions:
    test.ok(value, [message])
    test.equal(actual, expected, [message])
    test.notEqual(actual, expected, [message])
    test.deepEqual(actual, expected, [message])
    test.notDeepEqual(actual, expected, [message])
    test.strictEqual(actual, expected, [message])
    test.notStrictEqual(actual, expected, [message])
    test.throws(block, [error], [message])
    test.doesNotThrow(block, [error], [message])
    test.ifError(value)
*/

exports['awesome'] = {
  setUp: function(done) {
    // setup here
    done();
  },
  'detects bad words in a string': function(test) {
    test.expect(6);
    // tests here
    test.equal(typeof(wordfilter), 'object', 'should return an object');
    test.equal(wordfilter.blacklisted('this string contains the word skank'), true, 'skank should be true');
    test.equal(wordfilter.blacklisted('this string contains the word SkAnK'), true, 'SkAnK should be true');
    test.equal(wordfilter.blacklisted('this string contains the wordskank'), true, 'wordskank should be true');
    test.equal(wordfilter.blacklisted('this string contains the skankword'), true, 'skankword should be true');
    test.equal(wordfilter.blacklisted('this string is clean!'), false, 'should be false');
    test.done();
  },
  'add a word to blacklist': function(test) {
    wordfilter.addWords(['clean']);

    test.expect(1);
    test.equal(wordfilter.blacklisted('this string was clean!'), true, 'should be true');
    test.done();
  },
  'remove a single word from blacklist': function(test) {
    wordfilter.removeWord('crip');

    test.expect(1);
    test.equal(wordfilter.blacklisted('I have a prescription.'), false, 'should be false');
    test.done();
  },
  'clear blacklist': function(test) {
    wordfilter.clearList();

    test.expect(2);
    test.equal(wordfilter.blacklisted('this string contains the word skank'), false, 'list is now empty so false');

    wordfilter.addWords(['skank']);
    test.equal(wordfilter.blacklisted('this string contains the word skank'), true, 'skank was re-added so true');

    test.done();
  }
};

File: test/wordfilter_test.rb
$:.unshift(File.dirname(__FILE__) + '/../lib/')

require 'test/unit'
require 'wordfilter'

class WordfilterTest < Test::Unit::TestCase

	def setup
		super
		Wordfilter.init
	end

	def test_detects_bad_words_in_a_string
		assert(Wordfilter::blacklisted?("this string contains the word skank"), "Failed to mark a bad string as bad.");
		assert(Wordfilter::blacklisted?("this string contains the word SkAnK"), "Failed to mark a bad string as bad.");
		refute(Wordfilter::blacklisted?("this string was clean!"), "Failed to allow a non-bad string.");
	end

	def test_add_word_to_blacklist
		Wordfilter::add_words(['clean']);
		assert(Wordfilter::blacklisted?("this string was clean!"), "Failed to blacklist a string containing a new bad word.");
	end

	def test_added_words_checked_case_insensitively
		Wordfilter::add_words(['CLEAN']);
		assert(Wordfilter::blacklisted?("this string was clean!"), "Failed to blacklist a string containing a new bad word because of casing differences.");
	end

	def test_adding_scatological_words
		refute(Wordfilter::blacklisted?("we do not care about this shitty string"), "String was blacklisted when it should not have been.");
		Wordfilter::add_scatological_words();
		assert(Wordfilter::blacklisted?("we now care about this shitty string"), "Failed to blacklist a string containing a new bad word.");
	end

	def test_clear_blacklist
		Wordfilter::clear_list
		refute(Wordfilter::blacklisted?("this string contains the word skank"), "Cleared word list still blacklisting strings.");
		Wordfilter::add_words(['skank']);
		assert(Wordfilter::blacklisted?("this string contains the word skank"), "Failed to blacklist a string containing a new bad word.");
	end
end
